native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java

```diff
@@ -1232,7 +1244,15 @@ protected void setTime(PreparedStatement preparedStatement, String sqlType, int
                     } else {
                         preparedStatement.setTime(index, Time.valueOf(localTime));
                     }
-                } catch (Throwable e) {
+                } catch (Throwable ex) {
+                    if (ex instanceof SQLFeatureNotSupportedException) {
+                        if (zoneMap != null) {
+                            throw new TypeMismatchError("Unsupported type: offset of Civil is not supported by " +
+                                    "this database. " + ex.getMessage());
```

Comment:
what is the message we get in `ex.getMessage()` ?

Reply:
```Unsupported SQL type: TIMESTAMP_WITH_TIMEZONE```

Reply:
This logic can fail when the record has `zoneMap` and failed due to another supported type.